staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -473,6 +472,7 @@ func (qs *queueSet) removeTimedOutRequestsFromQueueLocked(queue *queue) {
 		queue.requests = reqs[removeIdx:]
 		// decrement the # of requestsEnqueued
 		qs.totRequestsWaiting -= removeIdx
+		metrics.ChangeRequestsInQueues(qs.qCfg.Name, fsName, -removeIdx)
```

Comment:
This doesn't seem right, how do you know fsName is about the request(s) being removed?

Reply:
I'd also expect a separate call per removal, since the requests are likely sourced by different FSs.

I think I'm claiming fsName should be stored with the request and not passed around.

Reply:
(I'm also claiming that the metrics ought to be about the request being added/ejected, not about the request triggering the ejection, hopefully we agree on that one)

Reply:
Oh!  Good catch, will fix.

Reply:
fixed.